### ISSUE TITLE
Formatters sometimes omit important info from parser error messages

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#9154]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 8.4.0
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

(no configuration)

**What did you do? Please include the actual source code causing the issue.**

I linted an invalid piece of JavaScript text which only consists of a single dot: `.`

```bash
$ echo '.' | eslint --stdin --no-eslintrc
```

**What did you expect to happen?**

I expected a parser error including the message `Unexpected token .`

**What actually happened? Please include the actual, raw output from ESLint.**

```
<text>
  1:1  error  Parsing error: Unexpected token
```

The `stylish` and `codeframe` formatters strip trailing `.` characters from report messages, presumably for stylistic reasons. However, this leads to confusing messages when the `.` is actually an important component of the message, rather than a period at the end of the sentence.

One way to address this would be to avoid removing the trailing `.` for `fatal` messages. Another solution would be to just stop removing trailing `.` characters entirely.

**Comment by i-ron-y** [comment]
How about keeping " ." but removing "." when there is no space in front of it? Seeing as that error message should've said "Unexpected token .", and the "." used as a punctuation in report messages would have some character that isn't a space immediately preceding it.

**Comment by i-ron-y** [comment]
PR'd.
